src/eth/storage/inmemory/inmemory_permanent.rs

```diff
@@ -353,6 +357,41 @@ impl PermanentStorage for InMemoryPermanentStorage {
 
         Ok(())
     }
+
+    async fn get_slots_sample(&self, start: BlockNumber, end: BlockNumber, max_samples: u64, seed: u64) -> anyhow::Result<Vec<SlotSample>> {
+        let state = self.lock_read().await;
+
+        let samples = state
+            .accounts
+            .iter()
+            .filter(|(_, account_info)| account_info.bytecode.get_current().is_some())
```

Comment:
Create a function `is_contract` in the account struct to make this filter more clear in its intent.